tests/valgrind/Makefile.BSD

```diff
@@ -5,6 +5,7 @@ PROG	=	potential-memleaks
 NOINST	=	1
 
 # Library code required
+LDADD_REQ	=	-lcrypto
```

Comment:
Hmm, we used to just have `LDADD_REQ +=` so anything already in `LDADD_REQ` would have stayed there.  I'm not sure that it matters, but we're now obliterating any value which was already set...

Reply:
Hmm, I just copied from `spiped/Makefile.BSD` and `spiped/Makefile.BSD`:
```
# Library code required
LDADD_REQ       =       -lcrypto
LDADD_REQ       +=      -lpthread
```

I'm happy to change it everywhere if you want.

Reply:
I think what we have here is good.  I can't see any reason for `LDADD_REQ` to be set anywhere other than in that block -- so if anything the problem is that we had `+=` there in the first place.  Maybe scan the trees to make sure the first `LDADD_REQ` is a `=` rather than a `+=`?

Reply:
Yeah, it's fine:

```
td@beastie: ~/src/spiped (test-valgrind)
$ git grep -B1 LDADD_REQ -- "**/Makefile.BSD"
spipe/Makefile.BSD-# Library code required
spipe/Makefile.BSD:LDADD_REQ    =       -lcrypto
spipe/Makefile.BSD:LDADD_REQ    +=      -lpthread
--
spiped/Makefile.BSD-# Library code required
spiped/Makefile.BSD:LDADD_REQ   =       -lcrypto
spiped/Makefile.BSD:LDADD_REQ   +=      -lpthread
--
tests/dnsthread-resolve/Makefile.BSD-# Library code required
tests/dnsthread-resolve/Makefile.BSD:LDADD_REQ  =       -lpthread
--
tests/pushbits/Makefile.BSD-# Library code required
tests/pushbits/Makefile.BSD:LDADD_REQ   =       -lpthread
--
tests/valgrind/Makefile.BSD-# Library code required
tests/valgrind/Makefile.BSD:LDADD_REQ   =       -lcrypto
tests/valgrind/Makefile.BSD:LDADD_REQ   +=      -lpthread
td@beastie: ~/src/spiped (test-valgrind)
$ 
```